ts/prime/Web/src/container/profile/changePassword.tsx

```
@@ -0,0 +1,51 @@
+import React, { useState } from 'react'
+import { default as Background } from '../../blocks/wrapper'
+import { withRouter } from 'react-router-dom'
+import { useAuthContext, useFormState } from '@/ApiReact'
+import { KeyValues, ContainerRouterProps } from '../../interface'
+import ChangePasswordForm from '../../components/user/changePassword'
+import SuccessComponent from '../../components/success'
+
+const ChangePassword: React.FC<ContainerRouterProps> = (props: ContainerRouterProps) => {
+  const useChangePasswordFormState = () => {
+    const initialValues: KeyValues = { oldPassword: '', newPassword: '' }
+    const auth = useAuthContext()
+
+    const { values, setValues, errors, setErrors, handleChange, handleSubmit } = useFormState({
+      initialValues,
+      handleSubmit: async (values: KeyValues): Promise<void> => {
+        try {
+          const response = await auth.changePassword(values.oldPassword, values.newPassword)
+          if (response.isOk()) {
+            setShowSuccess(true)
+          } else {
+            setErrors({ general: response.error.message })
+          }
+        } catch (e) {
+          console.log('===========CATCHED ERROR===========\n', e)
```

How do we want to handle this instead of just silently failing? What even is this failure case?

---

I wanted to set errors here to show it to user but i was never able to reach here. I need to access the errors to handle them.

---

so this would error if the `try` block fails, we should at least surface a general error right? I would even be ok with just dumping the error object `e`
We should eventually start logging all client side errors so that will help us uncover these hard to reach error states as time goes on